xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalOutgoingServerSession.java

```diff
@@ -261,6 +262,11 @@ static LocalOutgoingServerSession createOutgoingSession(@Nonnull final DomainPai
             // Wait for the future to give us a session...
             // Set a read timeout so that we don't keep waiting forever
             return (LocalOutgoingServerSession) sessionInitialiser.init(listener).get(INITIALISE_TIMEOUT_SECONDS.getValue().getSeconds(), TimeUnit.SECONDS);
+        } catch (NetworkEntityUnreachableException e) {
+            Log.warn("Cannot connect to XMPP domain '{}' (from '{}'): Unable to create a socket connection to a host that associated " +
+                "to the domain. This is typically caused by an outage (the remote server may be turned off) or a network configuration " +
+                "issue (eg: missing or invalid DNS records, restrictive firewalls, etc.).", domainPair.getRemote(), domainPair.getLocal());
```

Comment:
Hmmm, if the problem is actually a "missing or invalid DNS records", then that should be reported as the problem.  Why is this being speculated?  From Openfire's side, it is or it isn't, right?

Reply:
At this point in the code, we lost the exact cause of the problem. Also, I expect that there's a lot of grey area here. What if there _are_ DNS records, but not one for the actual server that's used? What if when there _is_ an A or AAAA record, but not a SRV record. If connection is failing, it's hard to deduce if that SRV record would've made a difference.